content/post/til-replace-module.md

```diff
@@ -0,0 +1,59 @@
+---
+title: "TIL: how to replace an external Go dependency with a local copy"
+date: 2021-02-05T14:54:34+01:00
+draft: false
+---
+
+## The Problem
+
+You're using a great open source library in your project, and then, one day, you find yourself thinking: 
+
+> "Oh, wouldn't be great if the author just added a tiny little log statement in that function?". 
+
+Yeah, it really would, this would allow me to trace the value of that variable or try a small change.
+
+What can you do: fork the library? Nope. Don't do that.
+
+
+## Replace it!
+
+You can temporarily replace the original module with a local copy, changing one line in your `go.mod` file.
+
+
+### Example
+
+Here's a minimal example of a small `go.mod`, the file every Go project uses to handle dependencies:
+
+```
+module github.com/zmoog/foo
+
+require (
+    github.com/zmoog/bar v0.1.0
+)
+```
+
+To replace the module `github.com/zmoog/bar` with a local copy located on your local disk, all you need to do is add the a new line with the [replace](https://golang.org/ref/mod#go-mod-file-replace) directive:
```

Comment:
all you need to do is add the a new line #

Reply:
Thank you! ❤️  Fixed in #3.